api/system.cc

```diff
@@ -158,6 +162,27 @@ void set_system(http_context& ctx, routes& r) {
             return json::json_return_type(json::json_void());
         });
     });
+
+    hs::dump_profile.set(r, [](std::unique_ptr<request> req) {
+        if (!__llvm_profile_dump) {
+            apilog.info("Profile will not be dumped, executable is not instrumented with profile dumping.");
+            return make_ready_future<json::json_return_type>(json::json_return_type(json::json_void()));
+        }
+        sstring profile_dest(__llvm_profile_get_filename ? __llvm_profile_get_filename() : "disk");
+        apilog.info("Dumping profile to {}", profile_dest);
+        __llvm_profile_dump();
+        if (__llvm_profile_reset_counters) {
+            // If counters are not reset the profile dumping mechanism will issue a warning and exit
+            // next time it is attempted. If the counters are reset, profiles can be accumulated
+            // (if %m is present in LLVM_PROFILE_FILE pattern) so it can be dumped in stages or
+            // multiple times during runtime.
+            __llvm_profile_reset_counters();
+        } else {
+            apilog.warn("Could not reset profile counters, profile dumping will be skipped next time it is attempted");
```

Comment:
I don't understand how this is right - in this "else" you don't actually do anything, so how will profile dumping get skipped next time? Maybe you're saying that the llvm_profile_dump() itself doesn't work when called the second time?